stock_inventory_revaluation/models/stock_inventory_revaluation_line.py

```diff
@@ -0,0 +1,192 @@
+        # account instead the stock valuation account
+        if self.stock_move_id._is_dropshipped():
+            debit_account_id = (
+                accounts.get("expense") and accounts["expense"].id or False
+            )
+        already_out_account_id = accounts["stock_output"].id
+        credit_account_id = (
+            self.account_id.id
+            or revaluation_product.categ_id.property_stock_account_input_categ_id.id
+        )
+
+        if not credit_account_id:
+            raise UserError(
+                _("Please configure Stock Expense Account for product: %s.")
+                % (revaluation_product.name)
+            )
+
+        return self._create_account_move_line(
+            move,
+            revaluation_line,
+            credit_account_id,
+            debit_account_id,
+            qty_out,
+            already_out_account_id,
+        )
+
+    def _prepare_base_line(self, move, revaluation_line):
+        base_line = {
+            "name": self.name,
+            "product_id": self.product_id.id,
+            "stock_move_id": move.id,
+            "stock_inventory_revaluation_line_id": revaluation_line.id,
+            "quantity": 0,
+        }
+        return base_line
+
+    def _create_account_move_line(
+        self,
+        move,
+        revaluation_line,
+        credit_account_id,
+        debit_account_id,
+        qty_out,
+        already_out_account_id,
+    ):
+        """
+        Generate the account.move.line values to track the inventory revaluation.
+        Afterwards, for the goods that are already out of stock, we should create
+        the out moves
+        """
+        AccountMoveLine = []
+        base_line = self._prepare_base_line(move, revaluation_line)
+        debit_line = dict(base_line, account_id=debit_account_id)
+        credit_line = dict(base_line, account_id=credit_account_id)
+        diff = self.additional_value
+        if diff > 0:
+            debit_line["debit"] = diff
+            credit_line["credit"] = diff
+        else:
+            # negative revaluation, reverse the entry
+            debit_line["credit"] = -diff
+            credit_line["debit"] = -diff
+        AccountMoveLine.append([0, 0, debit_line])
+        AccountMoveLine.append([0, 0, credit_line])
+
+        # Create account move lines for quants already out of stock
+        if qty_out > 0:
+            debit_line = dict(
+                base_line,
+                name=(self.name + ": " + str(qty_out) + _(" already out")),
+                quantity=0,
+                account_id=already_out_account_id,
+            )
+            credit_line = dict(
+                base_line,
+                name=(self.name + ": " + str(qty_out) + _(" already out")),
+                quantity=0,
+                account_id=debit_account_id,
+            )
+            diff = diff * qty_out / self.quantity
+            if diff > 0:
```

Comment:
```suggestion
            if diff > 0.0:
```
consider doing this change in all module

Reply:
Done, thanks.